### PR TITLE
URW-147: add ability to order t-shirts while paying for dues; +hotfixes

### PR DESCRIPTION
**Pull request by sebastian-king**

### CODE REVIEW

File: admin/add-ftp-user.php [hunk 1]
```diff
@@ -1,120 +1,173 @@
...
+                    VALUES
+                    (
+                     '" . $db->real_escape_string($email) . "',
+                     '" . $db->real_escape_string($token) . "'
+                    )
+                ");
+
+                if ($q) {
+                    email($email, "UNT Robotics FTP Access Registration",
+                        "Howdy, if you've received this e-mail, you already know what it's about: $link");
```
Comment (ntindle) [inline admin/add-ftp-user.php:68]: Lol

File: admin/add-ftp-user.php [hunk 2]
```diff
@@ -1,120 +1,173 @@
 <?php
 require('../template/top.php');
 
-if (isset($_POST)) {
+function handleUserDeletion($request) {
```
Comment (ntindle) [inline admin/add-ftp-user.php:4]: ? Intending add to have remove behavior

File: api/discord/bot.php [hunk 3]
```diff
@@ -105,6 +105,11 @@ public static function type($channel_id) {
 	public static function get_all_users($guild_id) {
 	    return static::send_api_request("/guilds/{$guild_id}/members?limit=1000");
     }
+
+    // utils
```
Comment (ntindle) [inline api/discord/bot.php:109]: Rude

File: cron/unassign-expired-good-standing.php [hunk 4]
```diff
@@ -8,30 +8,43 @@
...
+            foreach (DISCORD_GOOD_STANDING_DEPENDENT_ROLES as $role) {
+                AdminBot::remove_user_role($val->user->id, $role);
+            }
+
+            $id = AdminBot::create_dm($val->user->id);
+
+            do {
+                $result = AdminBot::send_message(
+                    "Hello! This is the UNT Robotics administrative bot.\n" .
+                    "Your _Good Standing_ role has expired and has been removed from your account. If you had any other roles in the server which required _Good Standing_, those have been removed as well.\n" .
```
Comment (ntindle) [inline cron/unassign-expired-good-standing.php:30]: Also dependent. List those roles as well so they know why they should pay

**Review by ntindle — APPROVED** [review]

File: admin/add-ftp-user.php (hunk 2 above)
Comment (sebastian-king) [reply admin/add-ftp-user.php:4]: What's your question exactly?

**Pull request merged**